tests/41end-to-end-keys/07-backup.pl

```diff
@@ -0,0 +1,271 @@
+my $fixture = local_user_fixture();
+
+my $current_version; # FIXME: is there a better way of passing the backup version between tests?
+
+test "Can create backup version",
+   requires => [ $fixture ],
+
+   proves => [qw( can_create_backup_version )],
+
+   do => sub {
+      my ( $user ) = @_;
+
+      do_request_json_for( $user,
+         method  => "POST",
+         uri     => "/unstable/room_keys/version",
+         content => {
+            algorithm => "m.megolm_backup.v1",
+            auth_data => "anopaquestring",
+         }
+      )->then( sub {
+         my ( $content ) = @_;
+         log_if_fail "Content", $content;
+
+         assert_json_keys( $content, "version" );
+
+         $current_version = $content->{version};
+
+         do_request_json_for( $user,
+            method  => "GET",
+            uri     => "/unstable/room_keys/version",
+         );
+      })->then( sub {
+         my ( $content ) = @_;
+         log_if_fail "Content", $content;
+
+         assert_json_keys( $content, "algorithm" );
+
+         assert_json_keys( $content, "auth_data" );
+
+         $content->{algorithm} eq "m.megolm_backup.v1" or
+            die "Expected algorithm to match submitted data";
+
+         $content->{auth_data} eq "anopaquestring" or
+            die "Expected auth_data to match submitted data";
+
+         # FIXME: check that version matches the version returned above
```

Comment:
hm, isn't this just comparing $current_version with $content->{version} or something?

Reply:
Yup, except that I don't think the version is returned yet, so I need to fix synapse to do that first. ;)

Reply:
So it turns out that Synapse *does* return the version, even though it wasn't in the proposal. :man_shrugging:

Reply:
oops. i can imagine that i fixed up some of the thinkos in the proposal whilst doing the implementation, but then given the implementation never got fully finished, the proposal wasn't updated - sorry.